lightning/src/ln/channel.rs

```diff
@@ -6334,6 +6335,10 @@ impl<SP: Deref> FundedChannel<SP> where
 				self.context.channel_state.set_our_tx_signatures_sent();
 			}
 
+			self.context.channel_state = ChannelState::AwaitingChannelReady(AwaitingChannelReadyFlags::new());
+			let need_channel_ready = self.check_get_channel_ready(0, logger).is_some()
+				|| self.context.signer_pending_channel_ready;
+			self.monitor_updating_paused(false, false, need_channel_ready, Vec::new(), Vec::new(), Vec::new());
```

Comment:
This is silly and obviously should change, but it's just to show that we need AwaitingChannelReady to have `MONITOR_UPDATE_IN_PROGRESS` set in order for `ChannelContext::unbroadcasted_funding` to give us our funding transaction in `ChannelManager::internal_tx_signatures`.

Without introducing interactive signing-specific flags to `AwaitingChannelReady` I'm not sure what the right way forward is here without messing V1 stuff up.

Reply:
Yeah we can do. It just means we'll not use the `unbroadcasted_funding` API method in this case. I guess that's fine.

Basically skips this logic: 

```rust
	fn if_unbroadcasted_funding<F, O>(&self, f: F) -> Option<O> where F: Fn() -> Option<O> {
		match self.channel_state {
			ChannelState::FundingNegotiated(_) => f(),
			ChannelState::AwaitingChannelReady(flags) =>
				if flags.is_set(AwaitingChannelReadyFlags::WAITING_FOR_BATCH) ||
					flags.is_set(FundedStateFlags::MONITOR_UPDATE_IN_PROGRESS.into())
				{
					f()
				} else {
					None
				},
			_ => None,
		}
	}
```